third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -55,6 +60,63 @@ WinUWPH264EncoderImpl::~WinUWPH264EncoderImpl() {
   Release();
 }
 
+namespace {
+HRESULT MakeMediaTypeOut(UINT32 width,
+                         UINT32 height,
+                         UINT32 target_bps,
+                         UINT32 frame_rate,
+                         IMFMediaType** res_out) {
+  HRESULT hr = S_OK;
+  // output media type (h264)
+  ON_SUCCEEDED(MFCreateMediaType(res_out));
+  auto media_type_out = *res_out;
+  ON_SUCCEEDED(media_type_out->SetGUID(MF_MT_MAJOR_TYPE, MFMediaType_Video));
+  ON_SUCCEEDED(media_type_out->SetGUID(MF_MT_SUBTYPE, MFVideoFormat_H264));
+  // Lumia 635 and Lumia 1520 Windows phones don't work well
+  // with constrained baseline profile.
+  // todo(fibann): this is obsolete and should probably use the profile passed
+  // in the settings
+  // ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_MPEG2_PROFILE,
+  // eAVEncH264VProfile_ConstrainedBase));
+
+  ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_AVG_BITRATE, target_bps));
+  ON_SUCCEEDED(media_type_out->SetUINT32(MF_MT_INTERLACE_MODE,
+                                         MFVideoInterlace_Progressive));
+  ON_SUCCEEDED(MFSetAttributeSize(media_type_out, MF_MT_FRAME_SIZE, width,
+                                  height));
+  ON_SUCCEEDED(MFSetAttributeRatio(media_type_out, MF_MT_FRAME_RATE,
+                                   frame_rate, 1));
```

Comment:
>frame_rate [](start = 35, length = 10)

How about having some check for validating frame_rate within a sane range? (1..140)?

Reply:
I'd rather avoid arbitrary limits. If there a way we can get the max framerate from MF/the encoder we can use that, otherwise I'd rather just test for 0 like the code is doing already.